creator-node/src/routes/files.js

```diff
@@ -55,19 +55,29 @@ const IMAGE_UPLOAD_IPFS_VERIFICATION_RETRY_COUNT = 5
 
 /**
  * Helper method to stream file from file system on creator node
- * Serves partial content using range requests
+ * Serves partial content if specified using range requests
+ * By default, checks path for file existence before proceeding
+ * If not provided, checks fs stats for path
  */
-const streamFromFileSystem = async (req, res, path) => {
+const streamFromFileSystem = async (
+  req,
+  res,
+  path,
+  checkExistence = true,
+  fsStats = null
+) => {
```

Comment:
both above params are defaulted to backwards-compat values but accepted so `getCID` can pass in values and prevent this function from making redundant disk checks

pls review this code carefully 🙏